src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery =
+                "SELECT c.id AS car_id, model, m.id as manufacturer_id, name, country "
+                        + "FROM cars c "
+                        + "JOIN manufacturers m "
+                        + "ON c.manufacturer_id = m.id "
+                        + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement
+                        = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get a list of drivers from DB.", e);
+        }
+        cars.forEach(car -> car.setDrivers(getDriversForCar(car.getId())));
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarWithManufacturerQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarWithManufacturerStatement
+                        = connection.prepareStatement(updateCarWithManufacturerQuery)) {
+            updateCarWithManufacturerStatement.setString(1, car.getModel());
+            updateCarWithManufacturerStatement.setLong(2, car.getManufacturer().getId());
+            updateCarWithManufacturerStatement.setLong(3, car.getId());
+            updateCarWithManufacturerStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car with id: "
+                    + car.getId() + ". ", e);
+        }
+        deleteRelationsCarsDrivers(car);
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String softDeleteCarQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement softDeleteCarStatement
+                        = connection.prepareStatement(softDeleteCarQuery)) {
+            softDeleteCarStatement.setLong(1, id);
+            return softDeleteCarStatement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id: " + id + ". ", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
```

Comment:
In this method get not only id, but all necessary info (except car Drivers) in the Query. Do not use get() in the end. Just set Drivers in each car using getDriversForCar() in the end.